tools/automated-dns-for-instances/main.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2020 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from google.cloud import dns
+import time
+import re
+
+PROJECT_ID=<PROJECT_ID>
+ZONE=<ZONE_ID>
+DOMAIN=<DOMAIN> # should have a trailing dot
+TTL=3600
+
+client = dns.Client(project=PROJECT_ID)
+zone = client.zone(ZONE, DOMAIN)
+
+def find_by_name(name):
+    records = zone.list_resource_record_sets()
+    for record in records:
+        if name in record.name:
+            rs = zone.resource_record_set(record.name, record.record_type, record.ttl, record.rrdatas)
+            return(rs)
+    return False
+
+def vmToDNS(event, context):
+    import base64, json
+    changes=zone.changes()
+    valid_statuses = ['STAGING','DELETED']
+
+    if 'data' in event:
```

Comment:
nit: be more specific to dictionary use `dict.get` to raise exception if event is string (rather than trying to match substring data)
```suggestion
    if event.get('data'):
```